src/Integration/Binding/SolutionRuleSetWriter.cs

```diff
@@ -66,15 +65,15 @@ public string WriteSolutionLevelRuleSet(string solutionFullPath, RuleSet downloa
         /// Generate a solution level rule set file path from the given <see cref="ProjectInformation"/>.
         /// </summary>
         /// <param name="ruleSetRootPath">Root directory to generate the full file path under</param>
-        /// <param name="sonarQubeProject">SonarQube project to generate a rule set file name path for</param>
+        /// <param name="sonarQubeProjectKey">SonarQube project key to generate a rule set file name path for</param>
         /// <param name="fileNameSuffix">Fixed file name suffix</param>
-        internal /* testing purposes */ static string GenerateSolutionRuleSetPath(string ruleSetRootPath, ProjectInformation sonarQubeProject, string fileNameSuffix)
+        internal /* testing purposes */ static string GenerateSolutionRuleSetPath(string ruleSetRootPath, string sonarQubeProjectKey, string fileNameSuffix)
         {
             Debug.Assert(fileNameSuffix != null);
 
             // Cannot use Path.ChangeExtension here because if the sonar project name contains
             // a dot (.) then everything after this will be replaced with .ruleset
-            string fileName = $"{PathHelper.EscapeFileName(sonarQubeProject.Key + fileNameSuffix)}.{FileExtension}";
+            string fileName = $"{PathHelper.EscapeFileName(sonarQubeProjectKey + fileNameSuffix)}.{FileExtension}";
```

Comment:
I'll have to update more tests, and it's not a bug rather than improvement. Please log a separate improvement bug.